core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      const customActionIcon = <Input.ActionButton name="custom" onClick={jest.fn()} />;
+      const { getByTestId, container } = render(
+        <Input name="test" size="tiny" icon="search" value="test value" actionIcon={customActionIcon} />
+      );
+
+      const leftIcon = container.querySelector('.Input-icon--left');
+      const actionButton = getByTestId('DesignSystem-Input-ActionButton');
+
+      expect(leftIcon).toBeInTheDocument();
+      expect(leftIcon).toHaveClass('mr-3');
+      expect(actionButton).toBeInTheDocument();
+    });
+  });
+
+  describe('Focus Behavior', () => {
+    it('can be focused programmatically', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      const focusSpy = jest.spyOn(input, 'focus');
+      input.focus();
+
+      expect(focusSpy).toHaveBeenCalled();
+      focusSpy.mockRestore();
+    });
+
+    it('maintains focus when explicitly focused', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      input.focus();
+
+      expect(input).toBeInTheDocument();
+    });
+  });
+
+  describe('Right Icon Wrapper Interactions', () => {
+    it('handles keyboard focus on right icon wrapper for info tooltip', () => {
+      const { container } = render(<Input name="test" info="Test info" />);
+
+      const rightIconWrapper = container.querySelector('[tabIndex="0"]');
+      expect(rightIconWrapper).toBeInTheDocument();
+
+      if (rightIconWrapper) {
+        fireEvent.focus(rightIconWrapper);
+        fireEvent.blur(rightIconWrapper);
+        expect(rightIconWrapper).toBeInTheDocument();
+      }
+    });
+
+    it('handles mouse interactions on right icon wrapper', () => {
+      const { container } = render(<Input name="test" info="Test info" />);
+
+      const rightIconWrapper = container.querySelector('[tabIndex="0"]');
+      expect(rightIconWrapper).toBeInTheDocument();
+
+      if (rightIconWrapper) {
+        fireEvent.mouseEnter(rightIconWrapper);
+        fireEvent.mouseLeave(rightIconWrapper);
+
+        fireEvent.mouseDown(rightIconWrapper);
+        fireEvent.mouseUp(rightIconWrapper);
+
+        expect(rightIconWrapper).toBeInTheDocument();
+      }
+    });
+
+    it('handles keyboard events on right icon wrapper', () => {
+      const { container } = render(<Input name="test" info="Test info" />);
+
+      const rightIconWrapper = container.querySelector('[tabIndex="0"]');
+      expect(rightIconWrapper).toBeInTheDocument();
+
+      if (rightIconWrapper) {
+        fireEvent.keyDown(rightIconWrapper, { key: 'Enter' });
+        fireEvent.keyDown(rightIconWrapper, { key: ' ' });
+        fireEvent.keyDown(rightIconWrapper, { key: 'Tab' });
+
+        expect(rightIconWrapper).toBeInTheDocument();
+      }
+    });
+  });
+
+  describe('Min Width Behavior', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.